File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> hash;

        for (int i = 0 ; i < nums.size() ; i++){
            if (hash.find(nums[i]) != hash.end()){ // exists already
                return {i, hash[nums[i]]};
            } else{
                hash[target-nums[i]] = i;
            }
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        
        # given a number , we know how much we should add to get to our target
        
        
        dict = {}
        
        for i , c in enumerate(nums):
            
            if target - c not in dict:
                dict[c] = i
                
            else: 
                return i, dict[target-c]
        